optimisation/design/Model_Optimisation_Design.md

```diff
@@ -528,6 +594,7 @@ If model optimisation is to be supported when running compiled MATLAB/Horace fro
 - Can compiled MATLAB call pure Python functions?
 	+ How should such functions be provided to MATLAB? As `module.function` strings, memory addresses, etc.?
 - Does compiled MATLAB have access to builtin MATLAB functions like `feval` and `str2func`?
+	+ Either way it can not run arbitrary m-files! No MATLAB user functions can be called using the Python interface.
 
```

Comment:
In principle yes... 

With code compiled using R2017b, all Matlab built-in functions and all included library functions are callable using:

```python
>>> import compiled_matlab_library
>>> m = compiled_matlab_library.initialize()
>>> m.eval('eig(rand(3))')
matlab.double([[-0.18794383321803343],[1.7526703425428873],[0.8398632587280312]])
```

However, newer versions of Matlab restrict access to only specifically denoted m-files. In principle all Matlab functions, including mex and built-ins, are still accessible within the Matlab process even for the compiled library but now we have to write a single-m-file wrapper class to allow arbitrary Python access to the library functions.

On the minus side this will be a pain but we would probably want a Python-side wrapper anyway to make the syntax nicer for users, so it's not *that* much pain... On the plus side it shouldn't matter what class types (value or handle) Horace/Herbert/PACE Matlab classes are since we will need to write a (handle) class wrapper for them anyway...

Reply:
I'll try to answer these issues another design document and will make PR to update the text here when the Matlab-Python interface is better understood.